### PR TITLE
plugin modernization

### PR DESCRIPTION
**Pull request by mawinter69**
The plugin is currently on a very old Jenkins version (1.565). This implies dependencies to other plugins that have been extracted from core since then and prevents me from uninstalling these plugins.

bump to require jenkins 2.361.4
use ionicons
replace some deprecated things with recommended alternatives  (e.g. javax.annotation)
fix optOut help for the job property
use newer maven in the test

Pane:
image

Retry link:
image

Dark Theme:
Pane:
image

Retry link:
image

### Submitter checklist

- [ ] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [ ] Ensure that the pull request title represents the desired changelog entry
- [ ] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

File: src/test/java/com/chikli/hudson/plugin/naginator/NaginatorCauseTest.java [hunk 1]
```diff
@@ -156,42 +158,44 @@ public void testCauseLinkWithLargeNumber() throws Exception {
...
     }
 
     @Issue("SECURITY-2946")
     @Test
-    public void testEscapedDisplayname() throws Exception {
+    public void testEscapedDisplayName() throws Exception {
         FreeStyleProject p = j.createFreeStyleProject();
         FreeStyleBuild build1 = p.scheduleBuild2(0).get();
-        build1.setDisplayName​("<div id=\"unescaped-displayname\">bad displayname</div>");
+        build1.setDisplayName("<div id=\"unescaped-displayname\">bad displayname</div>");
```
Comment (ikedam) [inline src/test/java/com/chikli/hudson/plugin/naginator/NaginatorCauseTest.java:192]: Change note: Unprintable whitespace between `setDisplayName` and `(` is removed.

File: src/main/resources/com/chikli/hudson/plugin/naginator/NaginatorOptOutProperty/help-optOut.html [hunk 2]
```diff
@@ -1,4 +1,4 @@
 <div>
-    By default, naginator offer a "rebuild" link to all failed builds. Some jobs anyway might not be designed to
+    By default, naginator offers a "retry" link to all failed builds. Some jobs anyway might not be designed to
```
Comment (ikedam) [inline src/main/resources/com/chikli/hudson/plugin/naginator/NaginatorOptOutProperty/help-optOut.html:2]: Change note: `retry` is correct: https://github.com/jenkinsci/naginator-plugin/blob/naginator-1.18.2/src/main/java/com/chikli/hudson/plugin/naginator/NaginatorRetryAction.java#L51

File: src/main/java/com/chikli/hudson/plugin/naginator/NaginatorActionFactory.java [hunk 3]
```diff
@@ -14,13 +14,18 @@
  * @author <a href="mailto:nicolas.deloof@gmail.com">Nicolas De Loof</a>
  */
 @Extension
-public class NaginatorActionFactory extends TransientBuildActionFactory {
+public class NaginatorActionFactory extends TransientActionFactory<AbstractBuild> {
```
Comment (ikedam) [inline src/main/java/com/chikli/hudson/plugin/naginator/NaginatorActionFactory.java:17]: Review note: `TransientBuildActionFactory` and `TransientActionFactory` is binary incompatible, but I believe OK
(OK means we don't need to change the major version to mark binary incompatible).
I believe external modules never depend on `NaginatorActionFactory`.

File: src/main/resources/com/chikli/hudson/plugin/naginator/NaginatorPublisher/config.jelly [hunk 4]
```diff
@@ -32,8 +32,7 @@
             </f:entry>
             <j:if test="${descriptor.isMatrixProject(it)}">
                 <f:entry title="${%How to apply the regular expression to matrix}" field="regexpForMatrixStrategy">
-                    <!-- unfortunatelly, f:enum doesn't support doCheckXxxx -->
-                    <f:select />
+                    <f:select default="TestParent"/>
```
Comment (ikedam) [inline src/main/resources/com/chikli/hudson/plugin/naginator/NaginatorPublisher/config.jelly:35]: context: https://github.com/jenkinsci/naginator-plugin/blob/naginator-1.18.2/src/main/java/com/chikli/hudson/plugin/naginator/NaginatorPublisher.java#L75

File: src/main/resources/com/chikli/hudson/plugin/naginator/NaginatorAction/badge.jelly [hunk 5]
```diff
@@ -1,2 +1,4 @@
 <?jelly escape-by-default='true'?>
-<img src="${resURL}/plugin/naginator/redo.png" title="${%Re-scheduled after failure}"/>
+<j:jelly xmlns:j="jelly:core" xmlns:st="jelly:stapler" xmlns:d="jelly:define" xmlns:l="/lib/layout" xmlns:t="/lib/hudson" xmlns:f="/lib/form">
+    <l:icon src="symbol-arrow-redo-outline plugin-ionicons-api" class="icon-sm"/>
```
Comment (ikedam) [inline src/main/resources/com/chikli/hudson/plugin/naginator/NaginatorAction/badge.jelly:3]: I want to preserve `title`. We can use `tooltip` for `<l:icon>`.
Would you see that works?

**Review by ikedam — CHANGES_REQUESTED** [review]
Thanks for the request.
The change almost looks good to me. I want to keep `title` for the icon if we could.

Could you attach screenshots for icons? I think I can try that in the next weekend (unfortunately, I don't have an appropriate environment now), but the screenshot will be helpful if that doesn't bother you.

File: src/main/resources/com/chikli/hudson/plugin/naginator/NaginatorPublisher/config.jelly (hunk 4 above)
Comment (mawinter69) [reply src/main/resources/com/chikli/hudson/plugin/naginator/NaginatorPublisher/config.jelly:35]: Without this when adding the Publisher for the first time an exception is thrown and the UI breaks.

File: src/main/resources/com/chikli/hudson/plugin/naginator/NaginatorAction/badge.jelly (hunk 5 above)
Comment (mawinter69) [reply src/main/resources/com/chikli/hudson/plugin/naginator/NaginatorAction/badge.jelly]: done

**Review by ikedam — APPROVED** [review]
Thanks for updates and screenshots!

**Comment by ikedam** [comment]
I'm planning makina a new release tomorrow: #60

**Pull request merged**